third_party/txt/src/txt/paragraph_txt.cc

```diff
@@ -427,8 +430,16 @@ bool ParagraphTxt::ComputeBidiRuns(std::vector<BidiRun>* result) {
       U16_GET(text_.data(), 0, bidi_run_start + bidi_run_length - 1,
               static_cast<int>(text_.size()), last_char);
       if (u_hasBinaryProperty(last_char, UCHAR_WHITE_SPACE)) {
-        has_trailing_whitespace = true;
-        bidi_run_count--;
+        // Check if the trailing whitespace occurs before the previous run or
+        // not. If so, this trailing whitespace was a leading whitespace.
+        int32_t second_last_bidi_run_start, second_last_bidi_run_length;
+        ubidi_getVisualRun(bidi.get(), bidi_run_count - 2,
+                           &second_last_bidi_run_start,
+                           &second_last_bidi_run_length);
+        if (bidi_run_start >= second_last_bidi_run_start) {
```

Comment:
Check whether `bidi_run_start == second_last_bidi_run_start + second_last_bidi_run_length`

`has_trailing_whitespace` will extend the second-to-last run by one character.  This only makes sense if the additional character is the whitespace character comprising the final run

Reply:
Oops, forgot to push -f, thought I had pushed the changes already!